apisix/discovery/kubernetes/schema.lua

```diff
@@ -28,113 +28,228 @@ local port_patterns = {
+                            oneOf = host_patterns,
+                        },
+                        port = {
+                            type = "string",
+                            default = "${KUBERNETES_SERVICE_PORT}",
+                            oneOf = port_patterns,
+                        },
+                    },
+                    default = {
+                        schema = "https",
+                        host = "${KUBERNETES_SERVICE_HOST}",
+                        port = "${KUBERNETES_SERVICE_PORT}",
+                    }
                 },
-                host = {
-                    type = "string",
-                    default = "${KUBERNETES_SERVICE_HOST}",
-                    oneOf = host_patterns,
+                client = {
+                    type = "object",
+                    properties = {
+                        token = {
+                            type = "string",
+                            oneOf = {
+                                { pattern = [[\${[_A-Za-z]([_A-Za-z0-9]*[_A-Za-z])*}$]] },
+                                { pattern = [[^[A-Za-z0-9+\/._=-]{0,4096}$]] },
+                            },
+                        },
+                        token_file = {
+                            type = "string",
+                            pattern = [[^[^\:*?"<>|]*$]],
+                            minLength = 1,
+                            maxLength = 500,
+                        }
+                    },
+                    default = {
+                        token_file = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+                    },
+                    ["if"] = {
+                        ["not"] = {
+                            anyOf = {
+                                { required = { "token" } },
+                                { required = { "token_file" } },
+                            }
+                        }
+                    },
+                    ["then"] = {
+                        properties = {
+                            token_file = {
+                                default = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+                            }
+                        }
+                    }
                 },
-                port = {
-                    type = "string",
-                    default = "${KUBERNETES_SERVICE_PORT}",
-                    oneOf = port_patterns,
+                default_weight = {
+                    type = "integer",
+                    default = 50,
+                    minimum = 0,
                 },
-            },
-            default = {
-                schema = "https",
-                host = "${KUBERNETES_SERVICE_HOST}",
-                port = "${KUBERNETES_SERVICE_PORT}",
-            }
-        },
-        client = {
-            type = "object",
-            properties = {
-                token = {
-                    type = "string",
+                namespace_selector = {
```

Comment:
What about reusing the namespace_selector schema?

Reply:
ok, I'll try to optimize the schema fully